### ISSUE TITLE
Add MAP_FIXED mode

### ISSUE THREAD

**Issue by danburkert** [danburkert/memmap-rs#21]
Add a construction option for creating `MAP_FIXED` memory maps at a known memory location. CC #20 .

**Comment by darfink** [comment]
I was thinking about implementing this but saw that `CreateFileMappingW` is used on Windows for `anonymous` maps. Is there any specific reason `VirtualAlloc` is not used instead?

**Comment by darfink** [comment]
Is it possible for this to be implemented in before the crate evaluation? Right now this is the only missing feature, preventing me from publishing my [detour](https://github.com/darfink/detour-rs) library.

I've been constrained to *retep998's* unpublished fork of [rust-mmap](https://github.com/retep998/rust-mmap). As previously mentioned I could go on and try to implement this myself but I believe there are some underlying reasons as to why `CreateFileMapping` is used on Windows (IPC?).

**Comment by dtolnay** [comment]
Is this something that can be added backward compatibly after 1.0?

**Comment by danburkert** [comment]
Yes, as I noted in https://github.com/danburkert/memmap-rs/pull/48#issuecomment-322954015 I think we can do this backwards compatibly post 1.0.
